### PR TITLE
Feature/scroll to top

### PR DESCRIPTION
**Pull request by KendraTang**
localhost_8080_(iPhone 6_7_8) (1)

### CODE REVIEW

File: packages/BackToTopButton/index.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+.float {
+  @extend %flex-center;
+  @extend %reset-button;
+
+  position: fixed;
+  background-color: $white;
+  right: $spacing-1;
+  bottom: $spacing-2;
+  width: $fs-36;
+  height: $fs-36;
```
Comment (tpai) [inline packages/BackToTopButton/index.vue]: I think `$fs-xx` is for text size, it does not make sense to put it with `width` and `height`.

@Bruce-Lan

File: styles/_variables.scss [hunk 2]
```diff
@@ -27,6 +27,7 @@ $fw-bold: 600;
 
 // font size
 $fs-40: 2.5rem;
+$fs-36: 2.25rem;
```
Comment (tpai) [inline styles/_variables.scss]: I am not sure about this, do we need `$fs-36` here?

@Bruce-Lan

File: packages/BackToTopButton/index.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <button
+    :class="[$style.float, { show: isShown }]"
+    @click="scrollToTop"
+    v-html="require('~/assets/ic-arror-top.svg')"
+  />
```
Comment (tpai) [inline packages/BackToTopButton/index.vue]: Why not just wrap existed Button component for this? I think to add a shape attribute for the round button is work.

```diff
 <Button
   style-type="default"
+  shape="round"
 >
  <img slot="icon" src="~/assets/ic-arror-top.svg">
 </Button>
```

Comment (KendraTang) [reply packages/BackToTopButton/index.vue]: The size of this float button is quite different from existed button. If we're going to use the existed `Button` component, we'll need to add 2 more options: `small`, `round`. So the usage would be like:
```vue
  <Button
    :class="[$style.float, { show: isShown }]"
    @click="scrollToTop"
    size="small"
    round
  >
    <span
      :class="$style.icon"
      slot="icon"
      v-html="require('~/assets/ic-arror-top.svg')"
    />
  </Button>
```
and overwrite the size of icon: 
```
.icon {
  width: $fs-16;
  height: $fs-16;
}
```

Comment (tpai) [reply packages/BackToTopButton/index.vue]: You can put the `isShown` flag out of Button component, like:

```jsx
<div :class="[{show: isShown}]">
  <Button round />
</div>
```

Then only one option which needs to add to the Button.

Overwriting icon size is ok to me, but I prefer to specify the px directly to the width and height.

File: packages/BackToTopButton/index.vue (hunk 1 above)
Comment (KendraTang) [reply packages/BackToTopButton/index.vue]: Personally prefer to set the width / height with `rem`, so the layout would change accordingly  to the font-size changes by the browser setting. Currently we're only using rem for font sizes, but using them to adjust layout is more responsive IMO.

Comment (KendraTang) [reply packages/BackToTopButton/index.vue]: If @Bruce-Lan insist to keep the values to specific `px` setting, it's okay for me to change them back

Comment (tpai) [reply packages/BackToTopButton/index.vue]: I have no doubt about the `rem` usage, just wondering the variable definition, maybe we can ask @Bruce-Lan about this.

Comment (KendraTang) [reply packages/BackToTopButton/index.vue]: Oh I Got it, it's just seems to be misleading to use `fs` as `width`. I'd prefer defining something like `$unit-16` to describe both (??). 
I'll use `rem` directly for now anyway 👌

**Pull request merged**